main/src/vtr_path_planning/src/mpc/speed_scheduler.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021, Autonomous Space Robotics Lab (ASRL)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * \file speed_scheduler.cpp
+ * \author Jordy Sehn, Autonomous Space Robotics Lab (ASRL)
+ */
+
+#include "vtr_path_planning/mpc/speed_scheduler.hpp"
+#include <algorithm>
+#include <iostream>
+
+
+double ScheduleSpeed(const std::vector<double>& disc_path_curvature_xy, const std::vector<double>& disc_path_curvature_xz_yz, double VF, int curr_sid, double planar_curv_weight, double profile_curv_weight, double eop_weight, double horizon_step_size)
+{
+
+    // Experimental Speed Scheduler:
+    // Takes in the desired forward_velocity and the pre-processed global path and reduces the set speed based on a range of tunable factors:
+    // 1. Planar Curvature
+    // 2. Profile Curvature
+    // 3. End of Path
+    // 4. Corridor Width (TODO)
```

Comment:
Are these still TODO

Reply:
Removing the TODO's 4 and 5 in next commit. I had implemented these but decided not to add them as during testing I didnt like the effect